framework/Models/Code.py

```diff
@@ -215,10 +225,11 @@ def _readMoreXML(self,xmlNode):
       if os.path.exists(abspath):
         self.preExec = abspath
       else:
+        self.foundPreExec = False
         self.raiseAMessage('not found preexec '+self.preExec,'ExceptedError')
     self.code = Code.CodeInterfaces.returnCodeInterface(self.subType,self)
     self.code.readMoreXML(xmlNode) #TODO figure out how to handle this with InputData
-    self.code.setInputExtension(list(a.strip('.') for b in (c for c in self.clargs['input'].values()) for a in b))
+    self.code.setInputExtension(list(a[0].strip('.') for b in (c for c in self.clargs['input'].values()) for a in b))
```

Comment:
?

Reply:
since ```delimiter``` is added, a list of tuple is used instead of list. The first element in the tuple is the file extension.